kasa/smart/smartdevice.py

```diff
@@ -645,3 +661,172 @@ async def set_fan_speed_level(self, level: int):
         if not self.is_fan:
             raise KasaException("Device is not a Fan")
         await cast(FanModule, self.modules["FanModule"]).set_fan_speed_level(level)
+
+    # Bulb interface methods
+
+    @property
+    def is_color(self) -> bool:
+        """Whether the bulb supports color changes."""
+        return "ColorModule" in self.modules
+
+    @property
+    def is_dimmable(self) -> bool:
+        """Whether the bulb supports brightness changes."""
+        return "Brightness" in self.modules
+
+    @property
+    def is_variable_color_temp(self) -> bool:
+        """Whether the bulb supports color temperature changes."""
+        return "ColorTemperatureModule" in self.modules
+
+    @property
+    def valid_temperature_range(self) -> ColorTempRange:
+        """Return the device-specific white temperature range (in Kelvin).
+
+        :return: White temperature range in Kelvin (minimum, maximum)
+        """
+        if not self.is_variable_color_temp:
+            raise KasaException("Color temperature not supported")
+
+        return cast(
+            ColorTemperatureModule, self.modules["ColorTemperatureModule"]
+        ).valid_temperature_range
+
+    @property
+    def has_effects(self) -> bool:
+        """Return True if the device supports effects."""
+        return "dynamic_light_effect_enable" in self._info
```

Comment:
Let's get rid of this from the interface.

Reply:
As per above comment re HA and effects maybe we should keep and implement.